### PR TITLE
Added habit model

### LINKED ISSUE #74: Create HabitModel
Spec out and implement an immutable model for instances of a habit object

### PR DESCRIPTION
**Pull request by loravocado**
## Description of Changes
Added the habit class

## Checklist

- [ ] Automated tests
- [ ] Screenshots included (if necessary)
- [ ] Code is well commented

Closes #74

### CODE REVIEW

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private Date lastCompleted;
+
+    HabitModel(String title, String reason, Date startDate) {
+        this.title = title;
+        this.reason = reason;
+        this.startDate = startDate;
+        //TODO: Get ID from Firestore
+    }
+
+    String getTitle() {
```
[the remark this answers is not in the document]
Comment (MacMalainey) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:20]: Probably best if we hide them behind getters even if we don't have setters.

[the remark this answers is not in the document]
Comment (j-vent) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:20]: Yeah June pointed that out to me in his PR, my bad.

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.Date;
+
+//TODO: Add Frequency
+public class HabitModel {
+    final private String title;
+    final private String ID;
+    final private String reason;
+    final private int streak;
+    final private Date startDate;
+    private Date lastCompleted;
```
Comment (MacMalainey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:11]: Why is this not final?

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    HabitModel(String title, String reason, Date startDate) {
```
Comment (MacMalainey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:13]: Why does the constructor not have all the possible fields?

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        this.title = title;
+        this.reason = reason;
+        this.startDate = startDate;
+        //TODO: Get ID from Firestore
```
Comment (MacMalainey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:17]: This model does not directly interface with Firestore, it should be given the ID by the component that creates it

**Review by MacMalainey — CHANGES_REQUESTED** [review]

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java (hunk 3 above)
Comment (loravocado) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:13]: Oh I was thinking like you could call a method that would increment the streak and set the last completed date?

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java (hunk 2 above)
Comment (loravocado) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:11]: if lastCompleted can be reassigned, then it shouldn't be final right?

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java (hunk 3 above)
Comment (loravocado) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:13]: actually w/e I'll just change it

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java (hunk 2 above)
Comment (loravocado) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:11]: actually w/e I'll just change it give me a sec

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Date;
+
+//TODO: Add Frequency
+public class HabitModel {
+    final private String title;
+    final private String ID;
+    final private String reason;
+    final private int streak;
+    final private Date startDate;
+    final private Date lastCompleted;
```
Comment (MacMalainey) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitModel.java:11]: No getter for lastCompleted

**Review by MacMalainey — CHANGES_REQUESTED** [review]

**Review by AtrazineC — APPROVED** [review]

**Review by j-vent — APPROVED** [review]
Looks good to me!

**Review by cheetiantey — APPROVED** [review]

**Pull request merged**